Dockerfile

```diff
@@ -1,58 +1,53 @@
-ARG GO_VERSION="1.21"
+ARG GO_VERSION="1.22"
 
 #--------------------------------------------#
 #--------Build KSOPS and Kustomize-----------#
 #--------------------------------------------#
+FROM --platform=${BUILDPLATFORM} golang:${GO_VERSION} AS base
+RUN apt update && apt install git make -y
+COPY go.* .
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
+COPY . .
 
-# Stage 1: Build KSOPS and Kustomize
-FROM golang:$GO_VERSION AS builder
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.6.1@sha256:923441d7c25f1e2eb5789f82d987693c47b8ed987c4ab3b075d6ed2b5d6779a3 AS xx
 
-ARG TARGETPLATFORM
-ARG PKG_NAME=ksops
+# Stage 1: Build KSOPS and Kustomize
+FROM --platform=${BUILDPLATFORM} base AS builder
+ARG TARGETPLATFORM \
+    TARGETARCH \
+    PKG_NAME=ksops
+COPY --link --from=xx / /
 
 # Match Argo CD's build
-ENV GO111MODULE=on
-ENV CGO_ENABLED=0
+ENV GO111MODULE=on \
+    CGO_ENABLED=0
 
 # Define kustomize config location
+ENV HOME=/root
 ENV XDG_CONFIG_HOME=$HOME/.config
 
-# Export templated Go env variables
-RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
-    export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) && \
-    export GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | cut -c2-)
-
-WORKDIR /go/src/github.com/viaduct-ai/kustomize-sops
-
-COPY . .
-RUN go mod download
-RUN make install
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    xx-go --wrap && \
```

Comment:
takes care of the plumbing like on line https://github.com/viaduct-ai/kustomize-sops/pull/271/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L21-L23

also has a verify command to virify the outcome of the binary.